### ISSUE TITLE
indigo.loadMolecule("ARS") crashes

### ISSUE THREAD

**Issue by twall** [epam/Indigo#162]
`std::out_of_range` is thrown and the process terminated.  Using 1.4.0b0 of epam.indigo (python bindings).

Indigo is configured like this:
```
            indigo = Indigo()
            indigo.setOption("ignore-stereochemistry-errors", True)
            indigo.setOption("standardize-keep-largest", True)
            indigo.setOption("standardize-charges", False)
            indigo.setOption("standardize-stereo", True)
            indigo.setOption("standardize-neutralize-zwitterions", True)
            indigo.setOption("standardize-clear-unusual-valences", True)
```

**Comment by IuriiPuzanov** [comment]
Hi Timothy,
   Thank you very much for using Indigo and finding the issue.
Could you please clarify what is expected result for such attempt to load this string as molecule?

Best Regards!
Iurii

**Comment by twall** [comment]
I’d maybe expect an exception, not a crash.  Ideally, I would have a
“is_valid_smiles” call in the indigo library, but I have to use
loadMolecule instead.  I’m using it as part of an auto suggest feature that
needs to accept a number of different encodings, and be able to provide
both a smiles lookup and a name lookup.  It’s actually a little problematic
that loadMolecule parses some common names in addition to compound
encodings.

**Comment by mkviatkovskii** [comment]
Hi @twall 
Could you please provide an example of molecule that crashes with `std::out_of_range`?
I have just tried to run this code:
```python
indigo = Indigo()
indigo.setOption("ignore-stereochemistry-errors", True)
indigo.setOption("standardize-keep-largest", True)
indigo.setOption("standardize-charges", False)
indigo.setOption("standardize-stereo", True)
indigo.setOption("standardize-neutralize-zwitterions", True)
indigo.setOption("standardize-clear-unusual-valences", True)
m = indigo.loadMolecule("ARS")
```
And it just raises the following exception:
```
Traceback (most recent call last):
  File "load_ars.py", line 14, in <module>
    m = indigo.loadMolecule("ARS")
  File "/home/mk/indigo-tests/indigo/api/python/indigo.py", line 2405, in loadMolecule
    return self.IndigoObject(self, self._checkResult(Indigo._lib.indigoLoadMoleculeFromString(string.encode(ENCODE_ENCODING))))
  File "/home/mk/indigo-tests/indigo/api/python/indigo.py", line 2347, in _checkResult
    raise IndigoException(Indigo._lib.indigoGetLastError())
indigo.IndigoException: molecule auto loader: SMILES loader: 'A' specifier is allowed only for query molecules
```
And if I try to run
```python
m = indigo.loadQueryMolecule("ARS")
```
Everything just works fine.

I completely agree, Indigo definitely should not crash process in any case. Also thanks for a good suggestion to add some method like `isSmilesValid`.

**Comment by twall** [comment]
After swapping out the native libraries for the latest available (1.4.0b6,
without upgrading the python package itself - I'm using 1.4.0b0), I no
longer see the crash.

I'll run some additional tests against some other variations to make sure.

T

**Comment by mkviatkovskii** [comment]
Dear @twall,
The new stable version of Indigo 1.4.1 is already in PyPI. Could you please check if the problem is gone?
